hazelcast/src/main/java/com/hazelcast/internal/blackbox/Blackbox.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.blackbox;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The Blackbox is responsible for recording all kinds of Hazelcast/JVM specific information to
+ * help out with all kinds of issues.
+ *
+ * Each HazelcastInstance has its own Blackbox.
+ *
+ * <h1>Duplicate Registrations</h1>
+ * The Blackbox is also lenient regarding duplicate registrations of sensors. So if a Sensor is created and
+ * an SensorInput is set and a new registration for the same Sensor is done, the old SensorInput is overwritten.
+ * The reason to be lenient is that the Blackbox should not throw exception. Of course there will be a log
+ * warning.
+ *
+ * <h1>Performance</h1>
+ * The Blackbox is designed for low overhead sensors. So once a sensor is registered, there is no overhead
+ * for the provider of the sensor data. The provider could have for example a volatile long field and increment
+ * this using a lazy-set. As long as the Blackbox can frequently read out this field, the Blackbox is perfectly
+ * happy with such low overhead sensor-inputs. So it is up to the provider of the sensor-input how much overhead
+ * is required.
+ */
+public interface Blackbox {
```

Comment:
Sure. I'll it to the documentation.

I got the term from the aeronautical industry when they are explaining the blackbox.